docs/resource-policy/policy/topology-aware.md

```diff
@@ -445,56 +445,102 @@ metadata:
 
 ## Controlling Topology Hints Via Annotations
 
-User is able mark certain pods and containers to have allowed or denied
-paths for mounts or devices. What this means is that when the system
-is generating topology hints, it will consult this allowed / denied path
-list to determine what hints are created. The deny path list is checked
-first and then the allowed path list. This means that you can deny all
-the mount/device paths and then allow only the needed ones for example.
-User can either set the path with "prefix" (this is the default) or with
-"glob" type. The "prefix" type means that the prefix of the mount/device
-path is checked for matches. The "glob" type means that user is able to
-put wildcards to the matched paths.
+It is possible to control whether and what kind of topology hints are
+generated using extra pod annotations. By default hints are generated
+from mounts and devices injected into the container. If pod resource API
+queries are enabled, query replies are also used for hint generation.
+
+### Enabling Or Disabling Selected Types of Topology Hints
+
+The `topologyhints.resource-policy.nri.io` annotation key can be used
+to enable or disable topology hint generation for one or more containers
+altogether, or selectively for mounts, devices, and pod resources types.
+More than one type can be specified as a comma-separated list. Additionally,
+the `all` and `none` types are recognized to mean all or none of these
+types. If no hint type annotations are present, all types of hints are
+enabled.
+
+For example:
+
+```yaml
+metadata:
+  annotations:
+    # disable topology hint generation for all containers by default
+    topologyhints.resource-policy.nri.io/pod: none
+    # enable mount-based hints for the 'diskwriter' container
+    topologyhints.resource-policy.nri.io/container.diskwriter: mounts
+    # enable device-based hints for the 'videoencoder' container
+    topologyhints.resource-policy.nri.io/container.diskwriter: devices
+    # enable pod resource-based hints for the 'dpdk' container
+    topologyhints.resource-policy.nri.io/container.dpdk: pod-resources
+    # enable device and pod resource-based hints for 'networkpump' container
+    topologyhints.resource-policy.nri.io/container.networkpump: devices,pod-resources
+```
+
+Note that for pod resource based hints, you also need to enable pod
+resource API queries using the corresponding configuration option, like
+this:
+
+```yaml
+apiVersion: config.nri/v1alpha1
+kind: TopologyAwarePolicy
+metadata:
+  name: default
+  ...
+spec:
+  ...
+  agent:
+    podResourceAPI: true
+```
+
+### Controlling Topology Hints by Path
+
+It is also possible to enable and disable topology hint generation based
+on mount or device path, using allow and deny lists. When the policy is
+tenerating topology hints, it consults these lists to decide whether hints
+for a particular mount or device is enabled. The denied list is consulted
```

Comment:
List names are "deny" and "allow". I'd suggest using these (perhaps quoted or `deny`/`allow`) in the sentence:

The denied list is consulted first, followed by the allow list.

(...or denied+allowed if you like it better, but perhaps not denied+allow)